proxy/proxy.go

```diff
@@ -1279,55 +1270,35 @@ func (p *Proxy) errorResponse(ctx *context, err error) {
 		copyHeader(ctx.responseWriter.Header(), perr.additionalHeader)
 	}
 
-	switch {
-	case code == 499:
-		req := ctx.Request()
-		remoteAddr := remoteHost(req)
-		uri := req.RequestURI
-		if i := strings.IndexRune(uri, '?'); i >= 0 {
-			uri = uri[:i]
-		}
-
-		p.log.Infof(
-			`client canceled after %v, route %s with backend %s %s%s, status code %d: %v, remote host: %s, request: "%s %s %s", user agent: "%s"`,
-			time.Since(ctx.startServe),
-			id,
-			backendType,
-			backend,
-			flowIdLog,
-			code,
-			err,
-			remoteAddr,
-			req.Method,
-			uri,
-			req.Proto,
-			req.UserAgent(),
-		)
-	default:
-		req := ctx.Request()
-		remoteAddr := remoteHost(req)
-		uri := req.RequestURI
-		if i := strings.IndexRune(uri, '?'); i >= 0 {
-			uri = uri[:i]
-		}
-
-		p.log.Errorf(
-			`error while proxying after %v, route %s with backend %s %s%s, status code %d: %v, remote host: %s, request: "%s %s %s", user agent: "%s"`,
-			time.Since(ctx.startServe),
-			id,
-			backendType,
-			backend,
-			flowIdLog,
-			code,
-			err,
-			remoteAddr,
-			req.Method,
-			uri,
-			req.Proto,
-			req.UserAgent(),
-		)
+	msgPrefix := "error while proxying "
+	logFunc := p.log.Errorf
+	if code == 499 {
+		msgPrefix = "client canceled "
+		logFunc = p.log.Infof
+	}
+	req := ctx.Request()
+	remoteAddr := remoteHost(req)
+	uri := req.RequestURI
+	if i := strings.IndexRune(uri, '?'); i >= 0 {
+		uri = uri[:i]
 	}
 
+	logFunc(
+		msgPrefix+`after %v, route %s with backend %s %s%s, status code %d: %v, remote host: %s, request: "%s %s %s", user agent: "%s"`,
```

Comment:
nit: we could name `msgPrefix` a `reason` and format it as any other argument like
```go
logf(
		`%s after %v, route %s with backend %s %s%s, status code %d: %v, remote host: %s, request: "%s %s %s", user agent: "%s"`,
		reason,
		time.Since(ctx.startServe),
		...)
```

Reply:
changed